crates/noirc_evaluator/src/ssa_refactor/ssa_gen/program.rs

```diff
@@ -17,10 +17,17 @@ pub(crate) struct Ssa {
 impl Ssa {
     /// Create a new Ssa object from the given SSA functions.
     /// The first function in this vector is expected to be the main function.
+    /// TODO: The above condition is somewhat hidden in the API, should we change it so that
+    /// TODO the caller needs to pass in the main function?
+    ///
     pub(crate) fn new(functions: Vec<Function>) -> Self {
         let main_id = functions.first().expect("Expected at least 1 SSA function").id();
+        // TODO: why is this sorting being done here?
+        // TODO: does the max_id need to be the main_id initially
+        // TODO or do generally just need the max_id from all functions?
         let mut max_id = main_id;
```

Comment:
There's no sorting being done here, we're just converting the Vec of functions into a btree_map so we can index the map by the function id of each function.

`max_id` isn't any specific id, it's just used to keep track of what the next free id would be in case we need to add a function to this Ssa struct later and that function needs a fresh ID not shared by existing functions. That is why we initialize `next_id: AtomicCounter::starting_after(max_id)` later on.